gnucash/report/reports/standard/invoice.scm

```diff
@@ -346,7 +346,7 @@ for styling the invoice. Please see the exported report for the CSS class names.
   (gnc:register-inv-option
    (gnc:make-multichoice-option
     (N_ "Display") (N_ "QR Code")
-    "te" "QR Code"
+    "te" (if (gnc-qrcode-available) "QR Code" "QR Code (not available on this system")
```

Comment:
>"QR Code (not available on this system"

S.B. "QR Code (not available on this system)"
but how about "QR Code (install libqrcode to enable this option)"

And of course both need to be marked for translation.

Reply:
I may be wrong; it's not sufficient to install libqrencode on the local system isn't it? The build must be initiated with libqrencode present?

Reply:
You're right, the message should say something like "rebuild with libqrcode to enable this option". That's not particularly useful for users. I'd suggest that it could be a mandatory dependency if you could accept qrencode 3.4.4; the requirement for >=4.0 locks out Ubuntu18.04. All of the other distros worth caring about [have at least 4.0](https://repology.org/project/qrencode/versions). OTOH it's kind of rude to stuff a new dependency in a stable release series.

Reply:
https://fukuchi.org/works/qrencode/#news 3.4.4 should be fine -- main benefit for 4.0.0 is memory efficiency. The win32/mac/flatpak build scripts could be amended in maint, and only suggested in Linux builds (and mandatory for 5.x).

Reply:
FWIW I would not make libqrcode a new dependency on maint.

As for the case of missing libqrcode,  I still think the option should not be present at all if gnucash is not built with libqrcode. It's missing in previous 4.x release as well. Considering we want backwards compatibility in stable releases, the code  should still work if the option is not defined.

Imagine someone creates an invoice with QR code enabled in gnucash 4.5 and loads that report in 4.4 (on another system for example). We want that report to still show though ignoring the QR code option. I think the whole 4.x series should behave that way. Though you could add a helpful message to the user starting from the 4.5 release. Such message should probably only be  shown if they run gnucash 4.5 without qr code support and the saved invoice had the option enabled. In all other situations either the option works as advertised (gnucash 4.5+ built with libqrcode) or it is as if the option never existed (gnucash 4.5+ built without libqrcode  and loaded report didn't have option set in the first place).
So that message depends on the build configuration and could suggest something like "Report was configured with QR code enabled but this edition of gnucash lacks support for it. Please contact your package maintainer to request QR code support."
The tricky part is to display it to the user without it being printed on the actual invoice. Perhaps make it @media: screen only ?

Reply:
The way it's currently written is imho "safe". Upgrading 4.4 to 4.5-with-qrencode will show new option working well. Now user saves such a report, and reverts to 4.4 will now have neither QR nor error. Upgrading to 4.5-without-qrencode and loading saved report will dump a small warning on tracefile but otherwise functional report. i.e. at worst a small warning if 4.5-without-qrencode encounters a saved-report requesting QR.

There's a printed QR error only if an invalid parameter exists eg IBAN > 34 characters. Or amount too large.

Whether the report options disable or hide QR option is another small UI issue. I have no particular preference.

Displaying QR warnings (missing QR library) on screen vs printed is an interesting trick.

Reply:
Oh, I agree it's  safe.  My comments were only a suggestion to slightly improve the user experience. Not mandatory...